src/game/SpellEffects.cpp

```diff
@@ -2936,37 +2942,6 @@ void Spell::EffectSummonPet(SpellEffectIndex eff_idx)
         return;
     }
 
-    Pet* NewSummon = new Pet;
-
-    // petentry==0 for hunter "call pet" (current pet summoned if any)
-    if (m_caster->GetTypeId() == TYPEID_PLAYER && NewSummon->LoadPetFromDB((Player*)m_caster, petentry))
-    {
-        if (NewSummon->getPetType() == SUMMON_PET)
-        {
-            // Remove Demonic Sacrifice auras (known pet)
```

Comment:
We used to do this check twice. I made sure we only do it once. I tested it, all warlock summons remove the aura, not one summon bypasses it, so.... It's removed redundancy if anything.